src/config/password.c

```diff
@@ -601,6 +601,13 @@ int run_performance_test(void)
 
 bool set_and_check_password(struct conf_item *conf_item, const char *password)
 {
+	// Check if the newly set password is the same as the old one
+	if(verify_password(password, config.webserver.api.pwhash.v.s, false) == PASSWORD_CORRECT)
```

Comment:
Do we also need to check if no password is set (or empty) to exit the check early?

Reply:
This is already checked here:

https://github.com/pi-hole/FTL/blob/2406e1a70ee9ef83b98ed672df2705d02b016da2/src/config/password.c#L611-L621

Reply:
I agree it's worthwhile to do this before investing any work in trying to validate a hash (it takes a lot of time). I have added and pushed this.